http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20170811_label.xml

```diff
@@ -1,6 +1,12 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
 <databaseChangeLog logicalFilePath="leonardo" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
     <changeSet logicalFilePath="leonardo" author="thibault" id="label">
+        <validCheckSum>8:1d8581dc0977ea88b1f006f6bc00f5b9</validCheckSum>
+        <comment>
+            Mysql 8.4+ does not allow partial keys to be referenced for a foreign key anymore, see https://bugs.mysql.com/bug.php?id=114838.
+            This changeSet has been modified to reflect that; the validCheckSum is the checksum from when the flag did not need to be set because the default was OFF.
+        </comment>
+        <sql> SET restrict_fk_on_non_standard_key = OFF; </sql>
```

Comment:
This retroactive change would allow this Liquibase to run on a brand new MySQL 8.4 – what happens for existing CloudSQL instances like Prod?

I wonder if we also need a new changeset that will run on those and idempotently set this flag.

Reply:
It does, but there is no rush to merge this, I can test mycloud sql locally too

Reply:
Everything works great on a dev clone locally. Adding the changeset would not impact the pre-existing database so this is safe to merge as is. I will bring up the issue of potentially drifting our DB config between Broad and Manifold at the tech lead round table though.

Reply:
Sounds good.

I wonder whether Liquibase sets it as a global or session variable. If session, then new & old instances wouldn't drift at all.

Reply:
It can be set as both, see https://dev.mysql.com/doc/refman/8.4/en/server-system-variables.html#sysvar_restrict_fk_on_non_standard_key

But in this case it is a session variable, because I did not set it as `SET GLOBAL`